src/exportMap/childContext.js

```diff
@@ -1,10 +1,18 @@
 import { hashObject } from 'eslint-module-utils/hash';
 
-let parserOptionsHash = '';
-let prevParserOptions = '';
+let optionsHash = '';
+let prevOptions = '';
 let settingsHash = '';
 let prevSettings = '';
 
+// Replacer function helps us with serializing the parser nested within `languageOptions`.
+function stringifyReplacerFn(_, value) {
+  if (typeof value === 'function') {
+    return value.toString();
```

Comment:
```suggestion
    return String(value);
```
identical, since it's a known function, but safer in general :-)

Reply:
Updated